### PR TITLE
DB-11216 Fix potential memory leaks caused by tc.openScan 

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2291/

**Comment by cloudspliceci** [comment]
TEST FAILED
https://jenkins.build.splicemachine-dev.io/job/AutomatedTesting/job/spliceengine-PR/134/

**Comment by cloudspliceci** [comment]
TEST FAILED
https://jenkins.build.splicemachine-dev.io/job/AutomatedTesting/job/spliceengine-PR/134/

**Comment by cloudspliceci** [comment]
TEST FAILED
https://jenkins.build.splicemachine-dev.io/job/AutomatedTesting/job/spliceengine-PR/134/

**Comment by cloudspliceci** [comment]
TEST FAILED
https://jenkins.build.splicemachine-dev.io/job/AutomatedTesting/job/spliceengine-PR/134/

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/IndexChanger.java [hunk 1]
```diff
@@ -403,322 +403,322 @@ private void doDelete()
...
+    {
+        closeIndexCC();
+        closeIndexSC();
+        if (rowHolder != null && !rowHolderPassedIn)
+        {
+            rowHolder.close();
+        }
+        baseCC = null;
+    }
 }
```
Comment (jyuanca) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/IndexChanger.java:724]: Only formatting changes for this file, or I missed anything?

**Review by jyuanca — APPROVED** [review]
nice cleanup

**Review by msirek — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Review by dgomezferro — APPROVED** [review]
Nice cleanup!

**Pull request merged**